load_latin_lemma.py

```diff
@@ -0,0 +1,56 @@
+import csv
+
+from tqdm import tqdm
+
+from lemmatization.models import Lemma
+
+
+file_path = "import-data/distinct_lemma_short_defs_frequencies.tsv"
+total_lines = sum(1 for i in open(file_path, "r"))
+print(f"Begin latin lemma import (size: {total_lines})")
+
+with open(file_path) as f:
+    csv_reader = csv.DictReader(f, delimiter="\t")
+    count = 0
+    batch_size = 5000
+    batch = []
+    all_lemmas_qs = Lemma.objects.values()
+    lemmas_data = {q["lemma"]: q for q in all_lemmas_qs}
+    for row in tqdm(csv_reader, total=total_lines):
+        count += 1
+        val = list(row.values())
+        lemma, alt_lemma, label, short_def, rank, count, rate = val
+        data = dict(lang="lat", lemma=lemma.lower())
+        lemma_exist = lemmas_data.get(lemma)
+        if alt_lemma:
+            data["alt_lemma"] = alt_lemma.lower()
+        else:
+            data["alt_lemma"] = lemma.lower()
+        if label:
+            data["label"] = label.lower()
+        else:
+            data["label"] = lemma.lower()
+        if rank:
+            data["rank"] = float(rank)
+        else:
+            data["rank"] = 99999
+        if count:
+            data["count"] = float(count)
+        else:
+            data["count"] = 0
+        if rate:
+            data["rate"] = float(rate)
+        else:
+            data["rate"] = 0.000
+        if not label:
+            data["label"] = lemma.lower()
+        # only adds to the batch lemma does not already exist in the database
+        if not lemma_exist and lemma:
```

Comment:
Probably not a big deal, but maybe this check could happen before the other fields get checked and set above? That way if it's not going to get added to the batch it doesn't get processed. Might shave some time off at scale.

Reply:
Yea, definitely makes sense to do the operations only if its being added to the batch.